docs/docker.md

```diff
@@ -103,6 +103,13 @@ introduction to Docker.
 
 -   Although default values **SHOULD** be defined, exposed ports **MUST NOT** be
     fixed, and **MUST** be configurable using `ENV` variables.
+    
+-   The `MAINTAINER` instruction is deprecated. The `LABEL` instruction is a much 
+    more flexible version of this and you should use it instead, as it enables 
+    setting any metadata you require, and can be viewed easily, for example 
+    with docker inspect. Dockerfiles **SHOULD** put the labels `maintainer`, 
+    `release-date`, `version`, and `vendor`.
```

Comment:
There needs to be a concrete example here to define what the various terms mean. For example:

* maintainer - the name of the caretaker organization
* gitrepo - where the source code can be found
* branch - the branch the image was built from
* release_date - the date the image was built.
* contacts - partial email addresses of responsible contacts  (e.g for security updates)
*  organization - partial email addresses of responsible contacts (to avoid full public addresses / spam)

```Dockerfile
ARG NODE_VERSION=8.16.1-slim
FROM node:${NODE_VERSION}
ARG GITHUB_ACCOUNT=fiware
ARG GITHUB_REPOSITORY=iotagent-ul
ARG DOWNLOAD=latest
ARG SOURCE_BRANCH=master
ARG BUILD_DATE=unknown

ENV DOWNLOAD=${DOWNLOAD}

LABEL maintainer="FIWARE Foundation e.V." \
  gitrepo=https://github.com/${GITHUB_REPOSITORY}/${GITHUB_ACCOUNT} \
  branch=${SOURCE_BRANCH} \
  release_date=${BUILD_DATE} \
  contacts=fernando.lopez,jason.fox \
  organization=fiware.org

CMD ["/hello"]
```

Note that a Docker [build hook](https://github.com/telefonicaid/iotagent-ul/blob/master/docker/hooks/build) can be used to *automate* the date and branch functions e.g.

```bash
docker build  --build-arg BUILD_DATE="$(date)" --build-arg SOURCE_BRANCH=$SOURCE_BRANCH -t $IMAGE_NAME .
```

But they must have defaults.

Finally include an example of how to 

```bash
$ docker inspect --format '{{ index .Config.Labels }}' <image-name>

map[branch:master contacts:fernando.lopez,jason.fox gitrepo:https://github.com/iotagent-ul/telefonicaid maintainer:FIWARE Foundation e.V. organization:fiware.org release_date:Fri Sep 20 10:40:47 CEST 2019]
```

This should also be present in the relevant Docker `README.md`

I can't see how `version` can be automated - it shouldn't be necessary as it should be part of the existing version tag and just adds maintenance overhead.

Reply:
The question about what should be mandatory and what should be optional, I think it should be resolved in the TSC as well